repository/src/main/java/org/alfresco/repo/workflow/activiti/ActivitiWorkflowEngine.java

```diff
@@ -242,45 +242,45 @@ public List<WorkflowInstance> cancelWorkflows(List<String> workflowIds)
     /**
     * {@inheritDoc}
     */
-    public WorkflowInstance cancelWorkflow(String workflowId)
-    {
-        String localId = createLocalId(workflowId);
-        try
-        {
-            ProcessInstance processInstance = runtimeService.createProcessInstanceQuery().processInstanceId(localId).singleResult();
-            if (processInstance == null)
-            {
-                throw new WorkflowException(messageService.getMessage(ERR_CANCEL_UNEXISTING_WORKFLOW));
-            }
-
-            // TODO: Cancel VS delete?
-            // Delete the process instance
-            runtimeService.deleteProcessInstance(processInstance.getId(), WorkflowConstants.PROP_CANCELLED);
-
-            // Convert historic process instance
-            HistoricProcessInstance deletedInstance = historyService.createHistoricProcessInstanceQuery().processInstanceId(processInstance.getId())
-                    .singleResult();
-            WorkflowInstance result = typeConverter.convert(deletedInstance);
-
-            // Delete the historic process instance
-            // MNT-15498
-            if (!activitiUtil.isRetentionHistoricProcessInstanceEnabled())
-            {
-                historyService.deleteHistoricProcessInstance(deletedInstance.getId());
-            }
-
-            return result;
-        }
-        catch (ActivitiException ae)
-        {
-            String msg = messageService.getMessage(ERR_CANCEL_WORKFLOW);
-            if (logger.isDebugEnabled())
-            {
-                logger.debug(msg, ae);
-            }
-            throw new WorkflowException(msg, ae);
-        }
-    }
+    public WorkflowInstance cancelWorkflow(String workflowId)
+    {
+        String localId = createLocalId(workflowId);
+        try
+        {
+            ProcessInstance processInstance = runtimeService.createProcessInstanceQuery().processInstanceId(localId).singleResult();
+            if (processInstance == null)
+            {
+                throw new WorkflowException(messageService.getMessage(ERR_CANCEL_UNEXISTING_WORKFLOW));
+            }
+
+            // TODO: Cancel VS delete?

```

Comment:
Deleting a running process instance cancels the process execution. So, here delete and cancel are the same thing.

Reply:
The method cancelWorkflow(String) and the TODO comment were already there before this PR. Git has assumed as new code but it isn't.